tests/multi-thread.cc

```diff
@@ -137,30 +138,41 @@ int main(int argc, char* argv[])
 		exit(1);
 	}
 
-	int n_threads = 10;
-	int niter = 500;
+	const int n_threads = 10;
+	const int niter = 500;
+	Parse_Options opts[n_threads];
 
 	printf("Creating %d threads, each parsing %d sentences\n",
 		 n_threads, niter);
 	std::vector<std::thread> thread_pool;
 	for (int i=0; i < n_threads; i++)
 	{
 		Dictionary dict = dicte;
-		if (0 == i%3) dict = dictr;
-
-		Parse_Options opts = optsa;
-		if (0 == i%2) opts = optsb;
```

Comment:
Ah hah! This is the bug (maybe): the intent was that the Russian dict would have spell-guessing disabled. And so the old otsb was supposed to always be used with russian. The russian dict was %3 but the opts was %2 and it also should have been %3.

It looks like you fix this ... does the test now always pass?

Reply:
This was not enough to always pass the test, but after using a separate `Parse_Option` per thread the test always passes.